File: programers/1-완주하지못한선수.py
#한명 빼고는 모두 완주한 상황
#해시
def solution(part, com):
    from collections import Counter
    part.sort()
    com.sort()
    result = Counter(part) - Counter(com)
    return list(result)[0]

'''
#시간초과
def solution(participant, completion):
    answer = participant.copy()
    for i in completion:
        answer.pop(answer.index(i))
        
    return ''.join(answer)
'''

File: programers/1-완주하지못한선수.js
function solution(participant, completion) {
    participant.sort();
    completion.sort();
    for(var i=0;i<participant.length;i++){
        if(participant[i] !== completion[i]){
            return participant[i];
        }

    }
}

// function solution(participant, completion) {
//     completion.forEach(element => {
//       participant.splice(participant.indexOf(element),1);
//     });
//   return participant[0];
// }